src/ApplicationSampler.cpp

```diff
@@ -183,11 +184,16 @@ namespace geopm
         m_sampler->check_sample_end();
 
         m_status->update_cache();
-        if (!m_is_first_update) {
+        if (m_is_first_update) {
+            for (int cpu_idx = 0; cpu_idx != m_num_cpu; ++cpu_idx) {
+                m_hint_last[cpu_idx] = m_status->get_hint(cpu_idx);
```

Comment:
Does it make sense to set ```m_hint_time``` on the first update as well?

Reply:
The current solution is to make m_hint_last a const map with a special initializer object:

https://github.com/geopm/geopm/pull/1582/files/f55967855a5a7b192bf0b768bb740b85eeed5377#diff-f3b7daf82cbdf6c5548dbd738a8faa5b015fb5477299011efc5672d89e6cf6feR122-R133

This has the benefit of telling the compiler that the number of keys in the map will not increase.